tests/models/biattentive_classification_network_test.py

```diff
@@ -0,0 +1,65 @@
+# pylint: disable=invalid-name,protected-access
+import pytest
+import numpy
+
+
+from allennlp.common.testing import ModelTestCase
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.params import Params
+from allennlp.models import Model
+
+
+class BiattentiveClassificationNetworkMaxoutTest(ModelTestCase):
+    def setUp(self):
+        super(BiattentiveClassificationNetworkMaxoutTest, self).setUp()
+        self.set_up_model('tests/fixtures/biattentive_classification_network/maxout_experiment.json',
+                          'tests/fixtures/data/sst.txt')
+
+    def test_bcn_can_train_save_and_load(self):
+        self.ensure_model_can_train_save_and_load(self.param_file)
+
+    def test_batch_predictions_are_consistent(self):
+        self.ensure_batch_predictions_are_consistent()
+
+    def test_forward_pass_runs_correctly(self):
+        training_tensors = self.dataset.as_tensor_dict()
+        output_dict = self.model(**training_tensors)
+        class_probs = output_dict['class_probabilities'][0].data.numpy()
+        numpy.testing.assert_almost_equal(numpy.sum(class_probs, -1),
+                                          numpy.ones(class_probs.shape[0]))
+
+    def test_mismatching_dimensions_throws_configuration_error(self):
+        params = Params.from_file(self.param_file)
+        # Make the encoder wrong - it should be 2 to match
+        # the embedding dimension from the text_field_embedder.
+        params["model"]["encoder"]["input_size"] = 10
+        with pytest.raises(ConfigurationError):
+            Model.from_params(self.vocab, params.pop("model"))
+
+
+class BiattentiveClassificationNetworkFeedForwardTest(ModelTestCase):
```

Comment:
No need to have a whole new class for this, with repeated tests.  You can just have a second `train_save_and_load` test in the main test class, like this: https://github.com/allenai/allennlp/blob/a69209232dec0abc7938a79f7f55fc5e237dc3ef/tests/models/encoder_decoders/wikitables/wikitables_semantic_parser_test.py#L19-L28